### PR TITLE
[Web] Redesign earn page balance block

### PR DESCRIPTION
**Pull request by rners01**
**Description:** Redesign earn page balance block
**Notion Task:** https://www.notion.so/saladtech/Redesign-Earn-page-balance-block-6994da3ca35a47c8bb9d2c91ee0bee66?pvs=25
**Notion Project:** https://www.notion.so/saladtech/Frontend-Earning-Page-Design-Refresh-7bc4991e6c5c47168d4450fa257ee9db

image
image

### CODE REVIEW

File: packages/web-app/src/modules/earn-views/components/BalanceStat.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const styles = (theme: SaladTheme) => ({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '16px',
+    fontFamily: 'Mallory',
+  },
+  title: {
+    color: '#D5DADF',
```
Comment (rners01) [inline packages/web-app/src/modules/earn-views/components/BalanceStat.tsx:13]: tbd with Nelson

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx [hunk 2]
```diff
@@ -1,73 +1,85 @@
...
+    if (count === 1) {
+      return `${count} hour`
+    }
+
+    return `${count} hours`
+  }
+
+  return (
+    <div>
+      <div className={classes.title}>Earning Summary</div>
```
Comment (Maks19) [inline packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:73]: Let's use `SectionHeader` here

File: packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx [hunk 3]
```diff
@@ -1,75 +1,82 @@
...
+          <SectionHeader>Pantry</SectionHeader>
+          <PantryContainer />
+        </div>
+      </Scrollbar>
+    </>
+  )
 }
 
-export const EarningSummaryPage = withLogin(withStyles(styles)(_EarningSummaryPage), EarningInformationPage)
+export const EarningSummaryPage = withLogin(withStyles(styles)(EarningSummaryPageRaw), EarningInformationPage)
```
Comment (Maks19) [inline packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx:82]: Why did you change it?

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx [hunk 4]
```diff
@@ -1,73 +1,85 @@
...
+      <p className={classes.subtitle}>Take a birds eye view on how you’ve used Salad to earn rewards.</p>
+      <div className={classes.row}>
+        <BalanceStat title={'Current Balance'} value={formatBalance(currentBalance)} />
```
Comment (Maks19) [inline packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:76]: You don't need {} in `title`

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx [hunk 5]
```diff
@@ -1,73 +1,85 @@
...
-                bonusEarningRate.multiplier
-              }x your normal earning rate. You have already earned ${formatBalance(
-                bonusEarningRate.earnedAmount,
-              )}/${formatBalance(bonusEarningRate.earnedAmountLimit)} of your bonus amount`}
-            />
-          )}
-        </div>
-      </div>
-    )
+    return `${count} rewards`
```
Comment (Maks19) [inline packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:57]: Can `count` be 0?

File: packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx [hunk 6]
```diff
@@ -1,11 +1,15 @@
 import { connect } from '../../connect'
 import type { RootStore } from '../../Store'
 import { EarningSummaryPage } from './pages/EarningSummaryPage'
+import { getNumberWithCommas } from './utils'
 
 const mapStoreToProps = (store: RootStore): any => ({
   currentBalance: store.balance.currentBalance,
   lifetimeBalance: store.balance.lifetimeBalance,
-  totalXp: store.xp.currentXp,
+  totalChoppingHours: store.xp.currentXp && getNumberWithCommas((store.xp.currentXp / 60).toFixed(0)),
```
Comment (Maks19) [inline packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx:9]: Let's move it to the variable. You can use conditional there and put 0 if `!store.xp.currentXp`

**Review by Maks19 — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx (hunk 5 above)
Comment (rners01) [reply packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:57]: yes

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx (hunk 2 above)
Comment (rners01) [reply packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:73]: it has totally different styles

File: packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx (hunk 3 above)
Comment (rners01) [reply packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx:82]: I changed it from class component to fucntional

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx (hunk 5 above)
Comment (rners01) [reply packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:57]: yes

File: packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx (hunk 6 above)
Comment (rners01) [reply packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx:9]: What is the pros to move it to  the variable?

File: packages/web-app/src/modules/earn-views/components/EarningSummary.tsx (hunk 2 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningSummary.tsx:73]: It looks pretty much the same for all headers

image

File: packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx (hunk 3 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/pages/EarningSummaryPage.tsx:101]: Yes, the naming
Why did you change the pattern with naming using an underscore?

File: packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx (hunk 6 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/EarningSummaryContainer.tsx]: I believe it should be clear for understanding. And you'll be able to put 0 in conditional to not check on undefined `totalChoppingHours`

**Comment by Maks19** [comment]
Closed as all changes in new PR - https://github.com/SaladTechnologies/salad-applications/pull/1012

**Pull request closed without merging**